pennylane/ftqc/parametric_midmeasure.py

```diff
@@ -0,0 +1,252 @@
+    def __init__(
+        self,
+        wires: Optional[Wires],
+        *,
+        angle: Optional[float],
+        plane: Optional[str],
+        reset: Optional[bool] = False,
+        postselect: Optional[int] = None,
+        id: Optional[str] = None,
+    ):
+        self.batch_size = None
+        super().__init__(wires=Wires(wires), reset=reset, postselect=postselect, id=id)
+        self.plane = plane
+        self.angle = angle
+
+    def _flatten(self):
+        metadata = (
+            ("angle", self.angle),
+            ("wires", self.raw_wires),
+            ("plane", self.plane),
+            ("reset", self.reset),
+            ("id", self.id),
+        )
+        return (None, None), metadata
+
+    # pylint: disable=arguments-renamed, arguments-differ
+    @property
+    def hash(self):
+        """int: Returns an integer hash uniquely representing the measurement process"""
+        fingerprint = (
+            self.__class__.__name__,
+            self.plane,
+            self.angle,
+            tuple(self.wires.tolist()),
+            self.id,
+        )
+
+        return hash(fingerprint)
+
+    def __repr__(self):
+        """Representation of this class."""
+        return f"{self._shortname}_{self.plane.lower()}(wires={self.wires.tolist()}, angle={self.angle})"
+
+    @property
+    def has_diagonalizing_gates(self):
+        """Whether there are gates that need to be applied for to diagonalize the measurement"""
+        return True
+
+    def diagonalizing_gates(self):
+        """Decompose to a diagonalizing gate and a standard MCM in the computational basis"""
+        if self.plane == "XY":
+            mat = np.array(
+                [[1, np.exp(-1j * self.angle)], [1, -np.exp(-1j * self.angle)]]
+            ) / np.sqrt(2)
+            return [qml.QubitUnitary(mat, wires=self.wires)]
+        if self.plane == "ZX":
+            return [qml.RY(-self.angle, self.wires)]
+        if self.plane == "ZY":
+            return [qml.RX(self.angle, self.wires)]
+
+        raise NotImplementedError(
+            f"{self.plane} plane not implemented. Available plans are 'XY' 'ZX' and 'ZY'."
+        )
+
+    def label(self, decimals=None, base_label=None, cache=None):  # pylint: disable=unused-argument
+        r"""How the mid-circuit measurement is represented in diagrams and drawings.
+
+        Args:
+            decimals=None (Int): If ``None``, no parameters are included. Else,
+                how to round the parameters.
+            base_label=None (Iterable[str]): overwrite the non-parameter component of the label.
+                Required to match general call signature. Not used.
+            cache=None (dict): dictionary that carries information between label calls
+                in the same drawing. Required to match general call signature. Not used.
+
+        Returns:
+            str: label to use in drawings
+        """
+        superscripts = {"X": "ˣ", "Y": "ʸ", "Z": "ᶻ"}
```

Comment:
I did not know we could do this --- for my own knowledge, how do we generate these ?

Reply:
I used this: https://lingojam.com/SuperscriptGenerator

Unfortunately, there doesn't seem to be a unicode superscript for capital X, Y or Z, so they have to be lowercase